source/channel_bootstrap.c

```diff
@@ -1166,14 +1188,11 @@ static void s_on_server_channel_on_shutdown(struct aws_channel *channel, int err
 
     channel_data->server_connection_args->shutdown_callback(
         server_bootstrap, error_code, channel, server_shutdown_user_data);
+    s_server_connection_args_release(channel_data->server_connection_args);
```

Comment:
I don't *think* there are bugs here, but let's move the release() down as far as we can, *right* before we delete channel_data.

Who knows if there's some critical data in there that the channel needs to destroy itself.